AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -3501,6 +3474,27 @@ - (ASLayout *)measureWithSizeRange:(ASSizeRange)constrainedSize
   return [self layoutThatFits:constrainedSize parentSize:constrainedSize.max];
 }
 
+- (void)setPreferredFrameSize:(CGSize)preferredFrameSize
+{
+  ASDN::MutexLocker l(__instanceLock__);
+
+  // Deprecated preferredFrameSize just calls through to set width and height
+  _style.width = ASDimensionMake(preferredFrameSize.width);
+  _style.height = ASDimensionMake(preferredFrameSize.height);
+  [self invalidateCalculatedLayout];
+}
+
+- (CGSize)preferredFrameSize
+{
+  ASDN::MutexLocker l(__instanceLock__);
+  
+  if (_style.width.unit == ASDimensionUnitPoints && _style.height.unit == ASDimensionUnitPoints) {
+    return CGSizeMake(_style.width.value, _style.height.value);
+  }
+
+  return CGSizeZero;
```

Comment:
It asserted previously, and I recommended we just return CGSizeZero. My rationale was that we should be lenient – users who are reading a deprecated property know that they need to update their code, and the last thing they want is an exception being thrown from an accessor.